src/Extensions/CatalogPageExtension.php

```diff
@@ -146,4 +146,14 @@ public function getSortFieldName()
             return $sortColumn ?: 'Sort';
         }
     }
+
+    /**
+     * Prevent object creation if no parents exist.
+     * @param \SilverStripe\Security\Member|null $member
+     * @return bool
+     */
+    public function canCreate($member)
+    {
+        return $this->getCatalogParents()->count() > 0;
```

Comment:
canCreate should return false only to deny create, but otherwise return null to avoid overriding base canCreate logic.